### ISSUE TITLE
Migrate the project to NextJs/React

### ISSUE THREAD

**Issue by akshaaatt** [akshitagupta15june/PetMe#141]
**Is your feature request related to a problem? Please describe.**
As the web app grows, we need to be able to maintain it efficiently, hence a framework for the same would be better.

**Describe the solution you'd like**
Using NextJs will make the app simpler while making sure it is future-secure.

**Describe alternatives you've considered**
We can stick to Vanilla Js and pure HTML as of now but that limits the learning and scope of this project.

**Additional context**
-

**Comment by akshitagupta15june** [comment]
@muditchoudhary and @Susmita-Dey what are your thoughts in this?

**Comment by Susmita-Dey** [comment]
> @muditchoudhary and @Susmita-Dey what are your thoughts in this?

Sounds good for the project but all credit will go to him after that. I mean to say that nothing would be left for what we did together to make the website.

**Comment by akshaaatt** [comment]
No @Susmita-Dey , I'll only structure the nextjs codebase and copy the work you guys have done.

**Comment by akshaaatt** [comment]
For example, we'll break the lengthy HTML file into chunks of small reusable react components.

**Comment by Susmita-Dey** [comment]
> For example, we'll break the lengthy HTML file into chunks of small reusable react components.

Then it's good.

**Comment by akshaaatt** [comment]
Could you guys turn this into a hacktoberfest issue?
We can have multiple PRs supporting this.

**Comment by akshaaatt** [comment]
Also, if we can have a branch on the repo which says `react` would be great. The react conversion will take some time hence we can't directly merge the upcoming PRs to main

**Comment by akshitagupta15june** [comment]
fine! go for it

**Comment by akshaaatt** [comment]
@akshitagupta15june just create a new branch called react. Only you can do so.

**Comment by akshitagupta15june** [comment]
feel free to create any issues

**Comment by d3vsuthar** [comment]
+1 for this. Let me know if you need help creating the components. Also highly recommend using Supabase or another JAMstack compatible database for the backend

**Comment by Mominadar** [comment]
Hi, I would also like to contribute to this effort.

**Comment by akshitagupta15june** [comment]
@akshaaatt please guide @Mominadar

**Comment by muditchoudhary** [comment]
> s the web app gr

Yes, he is correct. I think we need a framework to maintain the website efficiently and speed up the process of development.

**Comment by muditchoudhary** [comment]
> @muditchoudhary and @Susmita-Dey what are your thoughts in this?

Yes, he is correct. I think we need a framework to maintain the website efficiently and speed up the process of development.

**Comment by DugarRishab** [comment]
@akshaaatt, correct me if I am wrong but, I don't think the nextjs server will be compatible with a node js server. Just create a react app for now, and integrate the backend later. @muditchoudhary what do you think?

**Comment by DugarRishab** [comment]
> @DugarRishab , NextJs is a fullstack framework which provides its own backend and server. It is similar to an express app, just the way of doing it is different.

I know that is why said that it might not work. The current backend (yet to be accepted and merged) is a node server.

**Comment by muditchoudhary** [comment]
> @akshaaatt, correct me if I am wrong but, I don't think the nextjs server will be compatible with a node js server. Just create a react app for now, and integrate the backend later. @muditchoudhary @akshitagupta15june @Susmita-Dey what do you think?

I don't know much about them. But If next js has its own server and we are using node now how can we use both? I think it's not possible. It is sure we are going to use node for this one as we already have a PR.

**Comment by Anuran12** [comment]
@akshaaatt  Can I add a navbar and a footer on the react branch

**Comment by akshaaatt** [comment]
@Anuran12 I did add a TopBar to the branch but it would be great if you could use Material UI components to make it and convert. Please add the footer. Thanks!

**Comment by Anuran12** [comment]
@akshaaatt  Okay.. 

By the way. I have a Proposal. we are making the react branch from scratch. So would not it be better if we change the UI. Like first make a good UI on figma and then start to make the website according to the UI.

**Comment by muditchoudhary** [comment]
> art to make the website according to the UI.

yaa that's a good idea. Do you know Figma? and If yes, could you design the UI?

**Comment by Anuran12** [comment]
> > art to make the website according to the UI.
> 
> yaa that's a good idea. Do you know Figma? and If yes, could you design the UI?

I am learning. That's why i told you. And i would like to make a ui for this , it will give me a good practice. 😁✌️

**Comment by muditchoudhary** [comment]
> > > art to make the website according to the UI.
> > 
> > 
> > yaa that's a good idea. Do you know Figma? and If yes, could you design the UI?
> 
> I am learning. That's why i told you. And i would like to make a ui for this , it will give me a good practice. grinv

yaa go ahead. Awesome opportunity to learn it.
